src/MSBuild/XMake.cs

```diff
@@ -1427,17 +1532,29 @@ private static void ResetGatheringSwitchesState()
                             {
                                 if (graphBuildOptions != null)
                                 {
-                                    (result, exception) = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    graphResult = ExecuteGraphBuild(buildManager, graphBuildRequest);
+                                    ProjectGraphEntryPoint entryPoint = graphBuildRequest.ProjectGraphEntryPoints.First();
+                                    if (!entryPoint.GlobalProperties.ContainsKey(PropertyNames.IsGraphBuild))
+                                    {
+                                        entryPoint.GlobalProperties[PropertyNames.IsGraphBuild] = "true";
+                                    }
```

Comment:
I remember you fighting with this but I don't remember how we got to this point. Is it right to set it here? Is it right that the graph doesn't set it?

Reply:
@dfederm, feel free to opine, but my understanding is that it's not expected to be set for the entry point—the entry point is just created with the set of global properties known by XMake. The graph itself _does_ set it, and that means that when we get the result back, it has that property set as a global property...but then the set of known global properties per the entry point doesn't match the set that comes back from the result, so I'm adjusting that here.

Reply:
"The entry point doesn't match anything in the actual graph" is the part that feels wrong to me. Also in retrospect `ProjectGraphEntryPoint.GlobalProperties` should maybe be an `IReadOnlyDictionary` instead but alas.